create-fake-screenshot.js

```diff
@@ -1,56 +1,87 @@
-  ctx.quadraticCurveTo(20, 30, 30, 30);
-  ctx.stroke();
-
-  ctx.fillStyle = '#2c2d30'; // text color
-  var options = {
-    font: 'bold 24px Arial', // Lato
-    lineHeight: 1.375,
-    paddingX: 120,
-    paddingY: 20
-  };
-  CanvasTextWrapper(canvas, 'Hurr Slack', options);
-
-  ctx.fillStyle = '#9e9ea6'; // text color
-  options = {
-    font: '15px Arial', // Lato
-    lineHeight: 1.375,
-    paddingX: 250,
-    paddingY: 30
-  };
-  CanvasTextWrapper(canvas, time, options);
-
-  ctx.fillStyle = '#2c2d30'; // text color
-  options = {
-    font: '24px Arial', // Lato
-    lineHeight: 1.375,
-    paddingX: 120,
-    paddingY: 55
-  };
-  CanvasTextWrapper(canvas, text, options);
-  var imageData = canvas.toDataURL('image/png');
-  return imageData.substring(imageData.indexOf(',') + 1);
+module.exports = function createSlackImage (archiveUrl, trimmedText, callback) {
+  console.log('archive : ' + archiveUrl + 'trim:' + trimmedText);
+  phantom.create().then(ph => {
+    _ph = ph;
+    return _ph.createPage();
+  }).then(page => {
+    _page = page;
+    return page.property('viewportSize', { width: 320, height: 480 });
+  }).then(() => {
+    return _page.open(archiveUrl);
+  }).then(status => {
+    console.log(status);
+    // var username = env.vars.SLACK_USERNAME;
+    // var password = env.vars.SLACK_PASSWORD;
+    return _page.evaluate(function () {
+      $('#email').val('charlesv@gmail.com');
+      $('#password').val('xx');
+      $('#signin_btn').click();
+    });
+  }).then(wat => {
+    function checkHighlight () {
+      return _page.evaluate(function () {
+        return !!document.querySelector('.highlight');
+      });
+    }
+    var times = 0;
+    return new Promise((resolve, reject) => {
+      setTimeout(function waitForHighlight () {
+        checkHighlight().then(exists => {
+          if (exists) {
+            resolve(exists);
+          } else if (times++ > 9) {
+            console.log('giving up');
+            reject(Error('never found highlighted element'));
+          } else {
+            setTimeout(waitForHighlight, 1000);
+          }
+        });
+      }, 9000);
+    });
+  }).then(yey => {
+    return _page.evaluate(function () {
+      var message = document.querySelector('.highlight');
+      message.classList.remove('highlight');
+      message.scrollIntoView();
+      var messageBody = message.querySelector('.message_body');
+      messageBody.style.padding = '10px';
+      var box = messageBody.getBoundingClientRect();
+      var jqbox = $(messageBody).offset(); // only jquery gets this right, for some reason
+      return {
+        top: jqbox.top,
+        left: jqbox.left,
+        width: box.width,
+        height: box.height
+      };
+    });
+  }).then(rect => {
+    console.log('the rect', rect);
+    return _page.property('clipRect', rect);
+  }).then(() => {
+    return _page.render('out.png');
+  }).then(() => {
+    var imageData = require('fs').readFile('out.png', 'base64', callback);
```

Comment:
sorry, i just realized that you did this exactly right--you passed the callback from `createFakeScreenshot` right into `fs`. i guess the named dependency from `package.json` worked great. nevermind about my promise thing above.

this does mean that you don't need to return, since you're done--you can just `_page.close(); _ph.exit()` right here.